File: cs205/jpetroulesMPG/jpetroulesMPGcs/MainForm.cs
// MPG
// Jake Petroules
// 2010-09-10
// This program allows the user to calculate a vehicle's fuel efficiency by providing the amount of fuel used and the distance traveled

using System;
using System.ComponentModel;
using System.Windows.Forms;

namespace jpetroulesMPGcs
{
    public partial class MainForm : Form
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="MainForm"/> class.
        /// </summary>
        public MainForm()
        {
            this.InitializeComponent();
        }

        /// <summary>
        /// Ensures that the user enters a numeric value in the distance traveled text box.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="System.ComponentModel.CancelEventArgs"/> instance containing the event data.</param>
        private void distanceTraveledTextBox_Validating(object sender, CancelEventArgs e)
        {
            double number;
            bool valid = double.TryParse(this.distanceTraveledTextBox.Text, out number);

            this.distanceTraveledErrorProvider.SetError(this.distanceTraveledTextBox, valid && number > 0 ? string.Empty : "Please enter a number greater than 0.");
        }

        /// <summary>
        /// Ensures that the user enters a numeric value in the fuel used text box.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="System.ComponentModel.CancelEventArgs"/> instance containing the event data.</param>
        private void fuelUsedTextBox_Validating(object sender, CancelEventArgs e)
        {
            double number;
            bool valid = double.TryParse(this.fuelUsedTextBox.Text, out number);

            this.fuelUsedErrorProvider.SetError(this.fuelUsedTextBox, valid && number > 0 ? string.Empty : "Please enter a number greater than 0.");
        }

        /// <summary>
        /// Calculates the fuel efficiency and displays the result in the fuel efficiency text box.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        private void calculateButton_Click(object sender, EventArgs e)
        {
            double distanceTraveled = 0;
            double fuelUsed = 0;

            bool wasNumeric = double.TryParse(this.distanceTraveledTextBox.Text, out distanceTraveled) && double.TryParse(this.fuelUsedTextBox.Text, out fuelUsed);
            if (wasNumeric && distanceTraveled > 0 && fuelUsed > 0)
            {
                this.fuelEfficiencyTextBox.Text = (distanceTraveled / fuelUsed).ToString("N2");
            }
            else
            {
                this.fuelEfficiencyTextBox.Text = "Invalid";
            }
        }

        /// <summary>
        /// Resets the form, clearing all input, output and error providers.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        private void resetButton_Click(object sender, EventArgs e)
        {
            this.distanceTraveledTextBox.Clear();
            this.fuelUsedTextBox.Clear();
            this.fuelEfficiencyTextBox.Clear();
            this.distanceTraveledErrorProvider.Clear();
            this.fuelUsedErrorProvider.Clear();
        }
    }
}

File: cs205/jpetroulesMPG/jpetroulesMPGvb/MainForm.vb
' MPG
' Jake Petroules
' 2010-09-10
' This program allows the user to calculate a vehicle's fuel efficiency by providing the amount of fuel used and the distance traveled

Option Explicit On
Option Strict On

Public Class MainForm
    ''' <summary>
    ''' Ensures that the user enters a numeric value in the distance traveled text box.
    ''' </summary>
    ''' <param name="sender">The source of the event.</param>
    ''' <param name="e">The <see cref="System.ComponentModel.CancelEventArgs" /> instance containing the event data.</param>
    Private Sub distanceTraveledTextBox_Validating(ByVal sender As System.Object, ByVal e As System.ComponentModel.CancelEventArgs) Handles distanceTraveledTextBox.Validating
        Dim number As Double
        Dim valid As Boolean = Double.TryParse(Me.distanceTraveledTextBox.Text, number)

        Me.distanceTraveledErrorProvider.SetError(Me.distanceTraveledTextBox, If(valid And number > 0, String.Empty, "Please enter a number greater than 0."))
    End Sub

    ''' <summary>
    ''' Ensures that the user enters a numeric value in the fuel used text box.
    ''' </summary>
    ''' <param name="sender">The source of the event.</param>
    ''' <param name="e">The <see cref="System.ComponentModel.CancelEventArgs" /> instance containing the event data.</param>
    Private Sub fuelUsedTextBox_Validating(ByVal sender As System.Object, ByVal e As System.ComponentModel.CancelEventArgs) Handles fuelUsedTextBox.Validating
        Dim number As Double
        Dim valid As Boolean = Double.TryParse(Me.fuelUsedTextBox.Text, number)

        Me.fuelUsedErrorProvider.SetError(Me.fuelUsedTextBox, If(valid And number > 0, String.Empty, "Please enter a number greater than 0."))
    End Sub

    ''' <summary>
    ''' Calculates the fuel efficiency and displays the result in the fuel efficiency text box.
    ''' </summary>
    ''' <param name="sender">The source of the event.</param>
    ''' <param name="e">The <see cref="System.EventArgs" /> instance containing the event data.</param>
    Private Sub calculateButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles calculateButton.Click
        Dim distanceTraveled As Double = 0
        Dim fuelUsed As Double = 0

        Dim wasNumeric As Boolean = Double.TryParse(Me.distanceTraveledTextBox.Text, distanceTraveled) And Double.TryParse(Me.fuelUsedTextBox.Text, fuelUsed)
        If (wasNumeric And distanceTraveled > 0 And fuelUsed > 0) Then
            Me.fuelEfficiencyTextBox.Text = (distanceTraveled / fuelUsed).ToString("N2")
        Else
            Me.fuelEfficiencyTextBox.Text = "Invalid"
        End If
    End Sub

    ''' <summary>
    ''' Resets the form, clearing all input, output and error providers.
    ''' </summary>
    ''' <param name="sender">The source of the event.</param>
    ''' <param name="e">The <see cref="System.EventArgs" /> instance containing the event data.</param>
    Private Sub resetButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles resetButton.Click
        Me.distanceTraveledTextBox.Clear()
        Me.fuelUsedTextBox.Clear()
        Me.fuelEfficiencyTextBox.Clear()
        Me.distanceTraveledErrorProvider.Clear()
        Me.fuelUsedErrorProvider.Clear()
    End Sub
End Class
